src/internal/proc.js

```diff
@@ -12,15 +12,17 @@ export const TASK_CANCEL = {toString() { return '@@redux-saga/TASK_CANCEL' }}
 const matchers = {
   wildcard  : () => kTrue,
   default   : pattern => input => input.type === pattern,
-  array     : patterns => input => patterns.some(p => p === input.type),
+  stringable: pattern => input => input.type === pattern.toString(),
+  array     : patterns => input => patterns.some(p => (p.hasOwnProperty('toString') ? p.toString() : p) === input.type),
   predicate : predicate => input => predicate(input)
 }
 
 function matcher(pattern) {
   return (
-      pattern === '*'   ? matchers.wildcard
-    : is.array(pattern) ? matchers.array
-    : is.func(pattern)  ? matchers.predicate
+      pattern === '*'          ? matchers.wildcard
+    : is.array(pattern)        ? matchers.array
+    : is.stringableFn(pattern) ? matchers.stringable
```

Comment:
im wondering if we should restrict this to functions only, also as naming convention suggests (`is.func`) the name should be `is.stringableFunc`